src/Opdex.Platform.Application/EntryHandlers/Transactions/CreateNotifyUserOfTransactionBroadcastCommandHandler.cs

```diff
@@ -24,11 +24,16 @@ public async Task<bool> Handle(CreateNotifyUserOfTransactionBroadcastCommand req
         var transaction = await _mediator.Send(new RetrieveTransactionByHashQuery(request.TransactionHash, false), cancellationToken);
         if (transaction is not null) return false;
 
-        // find transaction on node
-        var sender = await _mediator.Send(new RetrieveCirrusUnverifiedTransactionSenderByHashQuery(request.TransactionHash), cancellationToken);
-        if (sender == Address.Empty) return false;
+        // find transaction on node, this isn't _really_ the sender, is the change output address, can throw not found if the transaction is not found
+        var changeAddress = await _mediator.Send(new RetrieveCirrusUnverifiedTransactionSenderByHashQuery(request.TransactionHash), cancellationToken);
 
-        await _mediator.Send(new MakeNotifyUserOfTransactionBroadcastCommand(sender, request.TransactionHash), cancellationToken);
+        // If a change address is found and it doesn't match the alleged sender, don't notify
+        if (changeAddress != Address.Empty && changeAddress != request.Sender) return false;
+
+        // If a change address wasn't found, trust the requests sender, notify anyways
+        var senderToNotify = changeAddress == Address.Empty ? request.Sender : changeAddress;
```

Comment:
Surely this would allow someone to send notifications to a logged in user. Unlikely to be a realistic problem, at least for the time we have this temporary solution.

Reply:
Yeah, we can't be 100% accurate right now. 
Transaction receipts from the full node only return for mined transactions. 
Raw transactions we can get from the mempool but can't deserialize everything atm to get the sender without further work either in our node repo, bringing in some stratis packages to here, or PR to the SFN repo.